src/main/java/com/fauna/stream/StreamRequest.java

```diff
@@ -10,37 +11,94 @@
             throw new IllegalArgumentException("token cannot be null or empty");
         }
+        if (cursor != null && startTs != null) {
+            throw new IllegalArgumentException("Only one of cursor, or start_ts can be set.");
+        }
     }
 
-    public StreamRequest(String token, String cursor) {
-        this.token = token;
-        this.cursor = cursor;
-        this.start_ts = null;
-        if (token == null || token.isEmpty()) {
-            throw new IllegalArgumentException("token cannot be null or empty");
-        }
+    public static StreamRequest fromTokenResponse(StreamTokenResponse tokenResponse) {
+        return new StreamRequest(tokenResponse.getToken(), null, null, null);
     }
 
-    public StreamRequest(String token, Long start_ts) {
-        this.token = token;
-        this.cursor = null;
-        this.start_ts = start_ts;
-        if (token == null || token.isEmpty()) {
-            throw new IllegalArgumentException("token cannot be null or empty");
+    public static class Builder {
+        final String token;
+        String cursor = null;
+        Long startTs = null;
+        Duration timeout = null;
+
+        /**
+         * Return a new StreamRequest.Builder instance with the given token.
+         * @param token     A Fauna Stream token.
+         */
+        public Builder(String token) {
+            this.token = token;
         }
+
+        /**
+         * Return the current Builder instance with the given cursor.
+         * @param cursor    A Fauna Stream cursor.
+         * @return          The current Builder instance.
+         * @throws          IllegalArgumentException If startTs has already been set.
+         */
+        public Builder cursor(String cursor) {
+            if (this.startTs != null) {
+                throw new IllegalArgumentException("only one of cursor, or startTs can be set.");
+            }
+            this.cursor = cursor;
+            return this;
+        }
+
+        /**
+         * Return the current Builder instance with the given start timestamp.
+         * @param startTs   A timestamp to start the stream at.
+         * @return          The current Builder instance.
+         * @throws          IllegalArgumentException If startTs has already been set.
+         */
+        public Builder startTs(Long startTs) {
+            if (this.cursor != null) {
+                throw new IllegalArgumentException("only one of cursor, or startTs can be set.");
+            }
+            this.startTs = startTs;
+            return this;
+        }
+
+        /**
+         * Return the current builder instance with the given timeout.
+         * This timeout is the HTTP client timeout that is passed to java.net.http.HttpRequest.Builder.
+         * By testing, it seems that the client waits for the first byte, not the last byte of the response.
+         * Therefore, it's a timeout on the stream being opened, but the stream can stay open indefinitely.
```

Comment:
What's the default, and if the default is null what does that mean? We should describe that clearly.

Also, to me, a query timeout and stream open timeout are essentially the same thing. Maybe we should use the same defaults if we're not already.